src/ptimec.hh

```diff
@@ -680,6 +680,46 @@ ftime_e(char* dst, off_t& off_inout, ssize_t len, const struct exttm& tm)
     PTIME_APPEND('0' + ((tm.et_tm.tm_mday / 1) % 10));
 }
 
+inline bool
+ptime_j(struct exttm* dst, const char* str, off_t& off_inout, ssize_t len)
+{
+    dst->et_tm.tm_yday = 0;
+    PTIME_CONSUME(1, {
+        if (str[off_inout] < '0' || str[off_inout] > '9') {
+            return false;
+        }
+        dst->et_tm.tm_yday = str[off_inout] - '0';
+    });
+    if (off_inout < len) {
+        if (str[off_inout] >= '0' && str[off_inout] <= '9') {
+            dst->et_tm.tm_yday *= 10;
+            dst->et_tm.tm_yday += str[off_inout] - '0';
+            off_inout += 1;
+        }
+    }
+    if (off_inout < len) {
+        if (str[off_inout] >= '0' && str[off_inout] <= '9') {
+            dst->et_tm.tm_yday *= 10;
+            dst->et_tm.tm_yday += str[off_inout] - '0';
+            off_inout += 1;
+        }
+    }
+
+    if (dst->et_tm.tm_yday >= 1 && dst->et_tm.tm_yday <= 366) {
```

Comment:
The documentation for [`tm_yday`](https://linux.die.net/man/3/ctime) shows the bounds being 0 to 365.